kratos/processes/calculate_discontinuous_distance_to_skin_process.cpp

```diff
@@ -374,26 +386,42 @@ namespace Kratos
         const std::vector<array_1d<double,3>>& rIntersectionPointsCoordinates)
     {
         const auto &r_geometry = rElement.GetGeometry();
-        const unsigned int n_cut_edges = rIntersectionPointsCoordinates.size();
+        const unsigned int n_cut_points = rIntersectionPointsCoordinates.size();
 
         // Get reference to ELEMENTAL_DISTANCES
         Vector& r_elemental_distances = rElement.GetValue(ELEMENTAL_DISTANCES);
 
         // If there are more than 3 (3D) or 2 (2D) intersected edges, compute the least squares plane approximation
         // using the ComputePlaneApproximation utility.
         // Otherwise, the distance is computed using the plane defined by the 3 (3D) or 2 (2D) intersection points.
-        const bool do_plane_approx = (n_cut_edges == TDim) ? false : true;
-
-        if (do_plane_approx){
-            // Call the plane optimization utility
+        const bool do_plane_approx = (n_cut_points == TDim) ? false : true;
+        const auto compute_plane_appoximation = [&] (const std::vector<array_1d<double,3>>& rPointVector) {
```

Comment:
Why the lambda?

Reply:
Same idea as before, small function that is used in the same method more than once

Reply:
Done!